File: Data Structures & Algorithms/climbing-stairs/submission-0.cpp
class Solution {
private:
    vector<int> dp;
public:
    int f(int n){
        if(dp[n]!=-1) return dp[n];
        return dp[n]=f(n-1)+f(n-2);
    }
    int climbStairs(int n) {
        dp = vector<int>(n+1,-1);
        dp[0]=1;dp[1]=1;
        return f(n);        
    }
};

File: Data Structures & Algorithms/maximum-product-subarray/submission-0.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        res=nums[0]
        curmin,curmax=1,1

        for num in nums:
            tmp=curmax*num
            curmax=max(tmp,num*curmin,num)
            curmin=min(tmp,num*curmin,num)
            res=max(res,curmax)
        return res
